packages/components/table/src/store/watcher.ts

```diff
@@ -202,6 +201,49 @@ function useWatcher<T>() {
       instance.emit('selection-change', newSelection)
     }
   }
+  const toggleRowStatus = <T>(
+    statusArr: T[],
+    row: T,
+    newVal: boolean
+  ): boolean => {
+    let changed = false
+    const index = statusArr.indexOf(row)
+    const included = index !== -1
+
+    const addRow = () => {
+      statusArr.push(row)
+      if (isArray(row.children)) {
+        row.children.forEach((item) => {
+          toggleRowSelection(item, newVal ?? !included)
+        })
```

Comment:
This causes the `select` and `selection-change` event to trigger several times

Reply:
so if a row having children was selected, the `select` and `selection-change` is expected to trigger only once?I do some surplus works because of it,it triggers only once at the beginning

Reply:
now it will trigger only once